naucse/models.py

```diff
@@ -145,6 +145,7 @@ def default_lesson_url(lesson, page='index'):
             'subpage_url': lambda page: lesson_url(lesson=lesson, page=page),
             'lesson': lesson,
             'page': self,
+            '$solutions': solutions,
```

Comment:
It makes the variable unusable from normal code. Could be any character that can't be used in variable names.
I'm not aware of any conventions, I just picked something.